include/powsybl/iidm/LoadingLimitsAdder.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_LOADINGLIMITSADDER_HPP
+#define POWSYBL_IIDM_LOADINGLIMITSADDER_HPP
+
+#include <powsybl/iidm/LoadingLimits.hpp>
+#include <powsybl/iidm/OperationalLimitsAdder.hpp>
+#include <powsybl/stdcxx/math.hpp>
+#include <powsybl/stdcxx/optional.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+class OperationalLimitsHolder;
+
+template <typename L, typename A>
+class LoadingLimitsAdder : public OperationalLimitsAdder<L> {
+public:
+    class TemporaryLimitAdder {
+    public:
+        TemporaryLimitAdder(OperationalLimitsHolder& owner, LoadingLimitsAdder<L, A>& parent);
+
+        LoadingLimitsAdder<L, A>& endTemporaryLimit();
+
+        TemporaryLimitAdder& ensureNameUnicity();
+
+        TemporaryLimitAdder& setAcceptableDuration(unsigned long duration);
+
+        TemporaryLimitAdder& setFictitious(bool fictitious);
+
+        TemporaryLimitAdder& setName(const std::string& name);
+
+        TemporaryLimitAdder& setValue(double value);
+
+    private:
+        void checkAndGetUniqueName();
+
+        bool nameExists(const std::string& name) const;
+
+    private:
+        OperationalLimitsHolder& m_owner;
+
+        LoadingLimitsAdder<L, A>& m_parent;
+
+        std::string m_name;
+
+        double m_value = stdcxx::nan();
+
+        stdcxx::optional<unsigned long> m_acceptableDuration;
+
+        bool m_fictitious = false;
+
+        bool m_ensureNameUnicity = false;
+    };
+
+public:
+    explicit LoadingLimitsAdder(OperationalLimitsHolder& owner);
+
+    LoadingLimitsAdder(const LoadingLimitsAdder&) = default;
+
+    LoadingLimitsAdder(LoadingLimitsAdder&&) noexcept = default;
+
+    ~LoadingLimitsAdder() noexcept override = default;
+
+    LoadingLimitsAdder& operator=(const LoadingLimitsAdder&) = default;
+
+    LoadingLimitsAdder& operator=(LoadingLimitsAdder&&) noexcept = default;
+
+    TemporaryLimitAdder beginTemporaryLimit();
+
+    bool hasTemporaryLimits() const;
+
+    A& setPermanentLimit(double limit);
+
+protected:
+    void checkLoadingLimits() const;
+
+private:
+    LoadingLimitsAdder<L, A>& addTemporaryLimit(const std::string& name, double value, unsigned long acceptableDuration, bool fictitious);
+
+protected:
+    OperationalLimitsHolder& m_owner;
+
+    double m_permanentLimit = stdcxx::nan();
+
+    LoadingLimits::TemporaryLimits m_temporaryLimits;
```

Comment:
To be discussed (again): Sonar says those attributes should be private and should be manipulated by using public accessors (see [here](https://sonarcloud.io/project/issues?id=com.powsybl%3Apowsybl-iidm4cpp&pullRequest=317&resolved=false&severities=CRITICAL&types=CODE_SMELL)) . To be Sonar compliant, I should : 
- revert the accessors removal 
- add one so that `m_owner` can be accessed from children classes 

=> Not convinced about thoses modifications... => to be discussed